worker.go

```diff
@@ -100,8 +131,29 @@ func startWorkers(fileName string, nbWorkers int, env PreparedEnv) error {
 						}
 					} else {
 						if c := l.Check(zapcore.ErrorLevel, "unexpected termination, restarting"); c != nil {
-							c.Write(zap.String("worker", absFileName), zap.Int("exit_status", int(fc.exitStatus)))
+							c.Write(zap.String("worker", absFileName), zap.Int("failure_count", failureCount), zap.Int("exit_status", int(fc.exitStatus)), zap.Duration("waiting", backoff))
 						}
+
+						upFunc.Do(func() {
+							backingOffLock.Lock()
+							defer backingOffLock.Unlock()
+							// if we end up here, the worker has not been up for backoff*2
+							// this is probably due to a syntax error or another fatal error
+							if failureCount >= maxConsecutiveFailures {
+								logger.Fatal("Worker had too many consecutive unexpected terminations", zap.String("worker", absFileName))
+								panic("Too many consecutive failures")
```

Comment:
```suggestion
```

This will never be called, `Fatal()` calls `os.Exit(1)`.

Maybe a better alternative is to just panic, which allows this library's consumers to recover, and Caddy will log the error anyway IIRC.